models.py

```diff
@@ -0,0 +1,130 @@
+"""SQLAlchemy models for Covid App"""
+
+from flask_bcrypt import Bcrypt 
+from flask_sqlalchemy import SQLAlchemy 
+
+db = SQLAlchemy()  
+bcrypt = Bcrypt()   
+ 
+def connect_db(app):    
+        db.app = app    
+        db.init_app(app)
+        
+class User (db.Model):
+    """User in the system"""
+
+    __tablename__ = "users" 
+
+    id = db.Column( 
+        db.Integer, 
+        primary_key=True    
+    )   
+    firstname = db.Column(  
+        db.Text,    
+        nullable=False  
+    )   
+    lastname = db.Column(   
+        db.Text,    
+        nullable=False  
+    )   
+    username = db.Column(   
+        db.Text,    
+        nullable=False, 
+        unique=True 
+    )   
+    email = db.Column(  
+        db.Text,    
+        nullable=False, 
+        unique=True 
+    )   
+    password = db.Column(   
+        db.Text,    
+        nullable=False  
+    )   
+    image = db.Column(  
+        db.Text 
+    )   
+    state = db.Column(  
+        db.Text,    
+        nullable=False  
+    )   
+    vax_date = db.Column(   
+        db.Text,
+        nullable=True  
+    )   
+    covid_status = db.Column(   
+        db.Text,
+        nullable=True
+    )   
+
+    review = db.relationship('Review', backref='users')
+
+    @classmethod
+    def signup(cls, firstname, lastname, username, 
+            email, password, image, state, vax_date, covid_status):
+        """Sign up user.
+
+        Hashes password and adds user to system.
+        """
+
+        hashed_pwd = bcrypt.generate_password_hash(password).decode('UTF-8')
+
+        user = User(
+            firstname=firstname,
+            lastname=lastname,
+            username=username,
+            email=email,
+            password=hashed_pwd,
+            image=image,
+            state=state,
+            vax_date=vax_date,
+            covid_status=covid_status)
+
+        db.session.add(user)
+        return user
+
+    @classmethod
+    def authenticate(cls, username, password):
+        """Find user with `username` and `password`.
+
+        If can't find matching user (or if password is wrong), returns False.
+        """
+
+        user = cls.query.filter_by(username=username).first()
+
+        if user:
+            is_auth = bcrypt.check_password_hash(user.password, password)
+            if is_auth:
+                return user
+                
+        return False
```

Comment:
@azaria-dedmon I don't see how this code block is related to creating a database and tables.